### PR TITLE
[IMP]8.0 product links migr

### PR DESCRIPTION
**Pull request by fevxie**
[INIT] Initialization migrate product_links module to 8.0

### CODE REVIEW

File: product_links/product_links.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+from openerp import models, fields
+
+
+class ProductLink(models.Model):
+    _name = 'product.link'
+    _rec_name = 'linked_product_id'
+
+    product_id = fields.Many2one(
+        comodel_name='product.template',
+        string='Soruce Product',
```
Comment (guewen) [inline product_links/product_links.py]: s/Soruce/Source/

File: product_links/product_links.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        required=True,
+        ondelete='cascade')
+    type = fields.Selection(
+        selection=[
+            ('cross_sell', 'Cross-Sell'),
+            ('up_sell', 'Up-Sell'),
+            ('related', 'Related')],
+        string='Link type',
+        required=True)
+    is_active = fields.Boolean('Active', defalut=True)
```
Comment (guewen) [inline product_links/product_links.py]: s/defalut/default/

File: product_links/product_links.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+
+class Product(models.Model):
```
Comment (guewen) [inline product_links/product_links.py]: s/Product/ProductTemplate/

File: product_links/product_links.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    _inherit = 'product.template'
```
Comment (guewen) [inline product_links/product_links.py]: Why did you move the links on the template? I disagree with this change. You may want to cross-sell/up-sell only one variant of a product to another.

:-1:

File: product_links/README.rst [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+e-commerce (for instance, it is used in magentoerpconnect).
+
+
+Credits
+=======
+
+Contributors
+------------
+
+* Qing Wang <wang.qing@elico-corp.com>
```
Comment (pedrobaeza) [inline product_links/README.rst]: Add here Guewen Baconnier

Comment (fevxie) [inline product_links/README.rst]: ok, I'll do it.

**Comment by guewen** [comment]
Thanks :+1:

File: product_links/__openerp__.py [hunk 6]
```diff
@@ -21,7 +21,7 @@
 
 {
     'name': 'Product Links',
-    'version': '7.0.0',
+    'version': '8.0.0',
```
Comment (pedrobaeza) [inline product_links/__openerp__.py]: This version is incorrect. It should be 1.0. Odoo will add 8.0 itself at the beginning.

**Comment by yvaucher** [comment]
@fevxie Your commits are not properly linked to you github account either you didn't configured git with an email or you need to add your email on your github account

**Comment by fevxie** [comment]
Hi @yvaucher, thanks for your reminder.  I config my email according https://help.github.com/articles/setting-your-email-in-git/ using commands: git config --global  && $GIT_COMMITTER_EMAIL && $GIT_AUTHOR_EMAIL,  I din't get any information about how to update this pull request email?  Do I need to update this?  

Thanks

**Comment by yvaucher** [comment]
@fevxie As you pushed those commit without email you need to amend them.

You can do this using

```
git rebase -i origin/8.0
```

replace all `pick` by `edit`
save and quit -> `:wq`

Then it will pause for each commit. Do:

```
git commit --amend --author="Author Name <email@address.com>"
git rebase --continue
```

Then push those changes here, you will need to force push:

```
git push -f
```

Now you should see you avatar on commits of this PR

**Comment by fevxie** [comment]
@yvaucher Thanks for your patient steps,  Done.

File: product_links/README.rst [hunk 7]
```diff
@@ -0,0 +1,36 @@
+a module adds links between products
+===========================================
```
Comment (guewen) [inline product_links/README.rst]: Should be "Product Links" and the `=` line should be of the sale length (otherwise the rst parser raises an error)

File: product_links/product_links.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        required=True,
+        ondelete='cascade',
+        default=lambda self: self.env.context.get('product_id', False))
+    linked_product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Linked product',
+        required=True,
+        ondelete='cascade')
+    type = fields.Selection(
+        selection=[
```
Comment (guewen) [inline product_links/product_links.py]: Could you keep the `get_link_type_selection` method for the selection?
There are 2 benefits:
- the existing modules that inherits this method will still be compatible
- the `selection_add` argument is limited because it does not allow, for instance, to choose the order of the selection items (which is a no-go for `state` fields at least).

Comment (fevxie) [inline product_links/product_links.py]: hi, @guewen  I didn't know why in the v7 there are two method  _get_link_type_selection for current model to call and another method get_link_type_selection for inherit model to inherit and override

File: product_links/README.rst (hunk 7 above)
Comment (fevxie) [inline product_links/README.rst]: what do you mean "should be of the sale length" ?  the same length with the sale module ?

File: product_links/product_links.py (hunk 8 above)
Comment (guewen) [inline product_links/product_links.py]: In v7, there were 2 methods because you couldn't put the method name as a string like in
`selection='_get_link_type_selection'`, it was a direct reference to a method. Thus, when you had to extend the selection, you had to entirely redefine the field. The indirection with 2 methods allowed to extend the second method.

Now, `_get_link_type_selection` is no longer required, you can directly call `get_link_type_selection`.

File: product_links/product_links.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    @api.model
+    def get_link_type_selection(self):
+        # selection can be inherited and extended
+        return [('cross_sell', 'Cross-Sell'),
+                ('up_sell', 'Up-Sell'),
+                ('related', 'Related')]
+
+    @api.model
+    def _get_link_type_selection(self):
+        return self.get_link_type_selection()
```
Comment (guewen) [inline product_links/product_links.py]: This one is useless in v8 (see https://github.com/OCA/e-commerce/pull/40#discussion_r31216690)

File: product_links/product_links.py (hunk 8 above)
Comment (fevxie) [inline product_links/product_links.py]: thanks for your explanation

**Comment by guewen** [comment]
Thanks! :+1:

**Comment by yvaucher** [comment]
@fevxie Thanks for the changes :+1: 

Will can merge once travis is happy
